no-vary-search.md

```diff
@@ -0,0 +1,340 @@
+
+#### … the Cache API
+
+The [Cache API](https://developer.mozilla.org/en-US/docs/Web/API/Cache) already respects the `Vary` header when doing its cache matching, per [special handling in its spec](https://w3c.github.io/ServiceWorker/#request-matches-cached-item). This proposal adds similar special handling for `No-Vary-Search`.
+
+The existing `ignoreVary` option to [`cache.match()`](https://developer.mozilla.org/en-US/docs/Web/API/Cache/match) will _not_ affect the processing of `No-Vary-Search`. `ignoreVary` is useful for increasing the default cache hit rate, by ignoring the requirements of the `Vary` header; having it decrease the default cache hit rate when applied to responses with `No-Vary-Search` headers would be surprising.
+
+The existing `ignoreSearch` option to `cache.match()` will override the effect of `No-Vary-Search`; that is, if `ignoreSearch` is specified, then `No-Vary-Search` is ignored.
+
+We could contemplate adding an `ignoreNoVarySearch` option to `cache.match()` in the future, but are not yet aware of any use cases for that level of customization.
+
+#### … other web platform caches
+
+There are several other URL-keyed caches on the web platform. The ones we are aware of are:
+
+- The [list of available images](https://html.spec.whatwg.org/#the-list-of-available-images)
+- [Module maps](https://html.spec.whatwg.org/#module-map)
+- [Shared workers](https://html.spec.whatwg.org/#shared-workers-and-the-sharedworker-interface:concept-sharedworkerglobalscope-constructor-url)
+- [Resources preloaded via `<link rel="preload">`](https://html.spec.whatwg.org/#map-of-preloaded-resources)
+
+The proposal plans to update cache key computation for these caches as well.
+
+Note that these caches are keyed by _request_ URL, whereas `No-Vary-Search` is a property of the _response_. See [below](#-for-other-browser-caches) for discussion on how this plays out.
+
+#### … HTTP caches
+
+The proposal augments [RFC 9111](https://httpwg.org/specs/rfc9111.html#constructing.responses.from.caches)'s notion of "Constructing Responses from Caches", providing a mechanism to relax the requirement of "The presented effective request URI \[sic] and that of the stored response match" in well-defined ways.
+
+Since everything about the HTTP cache is best-effort, using `No-Vary-Search` will not guarantee additional cache hits. As such, browsers might implement HTTP cache integration after this feature has already proven its worth on, e.g., preloading caches.
+
+### Navigated-to pages
+
+When navigating to a URL which can be matched with a cached resource via `No-Vary-Search`, we still treat the resource _as if_ it were fetched from the target URL, and not from the originally-cached URL. For example:
+
+- Service workers and resource timing APIs see fetches go by targeting the original request URL.
+- Once the document is constructed and displayed:
+  - `location.href` is the navigated-to-URL.
```

Comment:
maybe clarify this is after redirects, at least once?

Reply:
I'll just slip in "response URL"... I think getting too far into the details of redirects at this point risks drawing attention from the main point, which is to contrast two conceivable possibilities for the document's URL.